metaseq/hub_utils.py

```diff
@@ -592,7 +592,9 @@ def generate(
 
             logger.info(f"Preparing generator with settings {self.cfg.generation}")
             generator = self.task.build_generator(
-                self.models, self.cfg.generation, extra_gen_cls_kwargs={"stop": stop}
+                self.models,
+                self.cfg.generation,
+                extra_gen_cls_kwargs={"stop": stop, "need_logprobs": True},
```

Comment:
need log probs results in a lot more memory being used. we shouldn't use this unless we absolutely have to

Reply:
That's reasonable. How about do: 
```
if logprobs > 0:
    generator.need_logprobs = True
```

Reply:
Change applied.

Reply:
Perfect